src/System.IO.Pipes.AccessControl/tests/AnonymousPipeTests/AnonymousPipeServerStreamAclTests.cs

```diff
@@ -0,0 +1,138 @@
+        [InlineData(-1)]
+        public void Create_InvalidBufferSize(int bufferSize)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                CreateAndVerifyAnonymousPipe(PipeDirection.In, HandleInheritability.None, bufferSize, GetBasicPipeSecurity()).Dispose();
+            });
+        }
+
+        [Theory]
+        [InlineData(PipeDirection.In,  HandleInheritability.Inheritable, 0)]
+        [InlineData(PipeDirection.In,  HandleInheritability.Inheritable, 1)]
+        [InlineData(PipeDirection.In,  HandleInheritability.None, 0)]
+        [InlineData(PipeDirection.In,  HandleInheritability.None, 1)]
+        [InlineData(PipeDirection.Out, HandleInheritability.Inheritable, 0)]
+        [InlineData(PipeDirection.Out, HandleInheritability.Inheritable, 1)]
+        [InlineData(PipeDirection.Out, HandleInheritability.None, 0)]
+        [InlineData(PipeDirection.Out, HandleInheritability.None, 1)]
+        public void Create_ValidParameters(PipeDirection direction, HandleInheritability inheritability, int bufferSize)
+        {
+            CreateAndVerifyAnonymousPipe(direction, inheritability, bufferSize, GetBasicPipeSecurity()).Dispose();
+        }
+
+        public static IEnumerable<object[]> Create_CombineRightsAndAccessControl_MemberData() =>
+            from rights in Enum.GetValues(typeof(PipeAccessRights)).Cast<PipeAccessRights>()
+            from accessControl in new[] { AccessControlType.Allow, AccessControlType.Deny }
+            select new object[] { rights, accessControl };
+
+        // These tests match NetFX behavior
+        [Theory]
+        [MemberData(nameof(Create_CombineRightsAndAccessControl_MemberData))]
+        public void Create_CombineRightsAndAccessControl(PipeAccessRights rights, AccessControlType accessControl)
+        {
+            if (rights == PipeAccessRights.Synchronize && accessControl == AccessControlType.Deny)
+            {
+                Assert.Throws<ArgumentException>("accessMask", () =>
+                {
+                    PipeSecurity security = GetPipeSecurity(WellKnownSidType.BuiltinUsersSid, PipeAccessRights.Synchronize, AccessControlType.Deny);
+                });
+            }
+            else if ((rights == PipeAccessRights.FullControl || rights == PipeAccessRights.ReadWrite) && accessControl == AccessControlType.Allow)
+            {
+                VerifyValidSecurity(rights, accessControl);
+            }
+            else
+            {
+                PipeSecurity security = GetPipeSecurity(WellKnownSidType.BuiltinUsersSid, rights, accessControl);
+                Assert.Throws<UnauthorizedAccessException>(() =>
+                {
+                    AnonymousPipeServerStreamAcl.Create(PipeDirection.In, HandleInheritability.None, 0, security).Dispose();
+                });
+            }
+        }
+
+        [Theory]
+        [InlineData(PipeAccessRights.ReadWrite | PipeAccessRights.Synchronize, AccessControlType.Allow)]
```

Comment:
Why is this one separated out into its own test, but the Sychronize/Deny case above isn't?

Reply:
I am adding a comment to explain why we are handling Synchronize that way:
```
// When creating the PipeAccessRule for the PipeSecurity, the PipeAccessRule constructor calls AccessMaskFromRights, which explicilty removes the Synchronize bit from rights when AccessControlType is Deny
/ and rights is not FullControl, so using Synchronize with Deny is not allowed
```

Reply:
Also, I put it in a separate test because the `Create_CombineRightsAndAccessControl_MemberData` with all the MemberDatas is using `Enum.GetValues`, so I cannot add bitwise combos in a pretty way.

Reply:
Here: [corefx/blob/b49a8a9be1d53cd9e50cb68fd8540be25c65d433/src/System.IO.Pipes/src/System/IO/Pipes/PipeAccessRule.cs#L47](https://github.com/dotnet/corefx/blob/b49a8a9be1d53cd9e50cb68fd8540be25c65d433/src/System.IO.Pipes/src/System/IO/Pipes/PipeAccessRule.cs#L47)